app/celer-sim/Transporter.cc

```diff
@@ -203,6 +204,33 @@ void Transporter<M>::accum_action_times(MapStrDouble* result) const
         }
     }
 }
+//---------------------------------------------------------------------------//
+/*!
+ * Print progress after N events when requested.
+ */
+template<MemSpace M>
+void Transporter<M>::progress(EventId const id,
+                              size_type const num_primaries) const
+{
+    CELER_EXPECT(num_primaries > 0);
+
+    auto const id_val = id.unchecked_get();
+    std::string prim = (num_primaries == 1) ? " primary" : " primaries";
+    std::string msg = "Event " + std::to_string(id_val) + ": transporting "
+                      + std::to_string(num_primaries) + prim;
+
+    if (print_progress_)
+    {
+        if (id_val % print_progress_ == 0)
+        {
+            CELER_LOG_LOCAL(status) << msg;
+        }
+    }
+    else
+    {
+        CELER_LOG_LOCAL(status) << msg;
+    }
```

Comment:
Just do
```suggestion
    if (id_val % print_progress_ == 0)
    {
        CELER_LOG_LOCAL(status) << msg;
    }
```
and replace `msg` with all the string stuff you did above. Use `(num_primaries == 1 ? " primary" : " primaries")` to avoid making a temporary string.

I'd also put the message writing up in the anonymous namespace (there's no need for templating) and put the `if (id_val % print_progress_ == 0) print_progress();` in the transporter function itself.

Reply:
I see here why you suggested `print_progress` to be always positive above...